tasks/Task.cpp

```diff
@@ -194,3 +194,18 @@ void Task::cleanupHook()
 {
     TaskBase::cleanupHook();
 }
+
+bool Task::cancelCurrentTrajectory()
+{
+    LOG_INFO_S << "Current trajectory cancellation requested";
+    // try to remove the trajectory and if no exception occurs, return true
+    try {
+        trajectoryFollower.removeTrajectory();
+    } catch (std::runtime_error& e) {
+        LOG_ERROR_S << "Could not remove trajectory: " << e.what();
+        return false;
+    }
+    trajectoryFollower.removeTrajectory();
```

Comment:
Is it intentional that removeTrajectory() is called twice here?

Reply:
No, That is indeed a mistake.